rb_tree.py

```diff
@@ -523,3 +534,100 @@ def _get_sibling(self, node):
             sibling = parent.right
             direction = 'R'
         return sibling, direction
+
+    def _build_tree_string(self, root=None, curr_index=0, index=False, delimiter='-'):
+    
+        if curr_index == 0:
+            root = self.root
+            
+        if root is None:
+            return [], 0, 0, 0
+
+        line1 = []
+        line2 = []
+        
+        if index:
+            node_repr = '{}{}{}'.format(curr_index, delimiter, root.value)
+        else:
+            node_repr = str(root.value)
```

Comment:
nit: Mind just doing `'NIL' if root.value is None else str(root.value)`? Results in more succinct code

Reply:
Done! Check it out please.